aten/src/ATen/native/LossCTC.cpp

```diff
@@ -210,7 +210,8 @@ Tensor ctc_loss_backward_cpu_template(const Tensor& grad_out, const Tensor& log_
   at::parallel_for(0, batch_size, 0, [&](int64_t start, int64_t end) {
     for (int64_t b = start; b < end; b++) {
       scalar_t nll = neg_log_likelihood.accessor<scalar_t, 1>()[b];
-      if (zero_infinity &&  nll == std::numeric_limits<scalar_t>::infinity()) {
+      if (zero_infinity &&
+          (nll == std::numeric_limits<scalar_t>::infinity() || std::isnan(nll))) {
```

Comment:
You mention the gradient becoming NaN, yet here you add a condition for `nll`. Have you actually observed `nll` becoming NaN? What were the inputs when that happened? Is this deterministic?

Reply:
No, i never observe NaN in CPU kernel. This part of code is just to make `zero_infinity` behaves in the same manner as the GPU part. And I am OK to remove it.